File: Excel/481 Challenge.R
library(tidyverse)
library(readxl)
library(tictoc)

path = "Excel/481 Taxicab Numbers.xlsx"
input = read_excel(path, range = "A1:A10")
test  = read_excel(path, range = "B1:B10")

tic()
is_taxicab = function(number) {
  x = ceiling(number^(1/3))

  df = tibble(a = 1:x, b = 1:x) %>%
    expand.grid() %>%
    filter(a <= b,
           a^3 + b^3 == number)
  check = ifelse(nrow(df) >= 2, "Y", "N")
  return(check)
}

result = input %>%
  mutate(`Answer Expected` = map_chr(Numbers, is_taxicab)) 
toc()
# 0.03 sec elapsed

identical(result$`Answer Expected`, test$`Answer Expected`)
# [1] TRUE

File: Excel/400-499/481/481 Challenge.py
import pandas as pd
import itertools
import time

path = '481 Taxicab Numbers.xlsx'
input = pd.read_excel(path, usecols="A")
test  = pd.read_excel(path, usecols="B")

start_time = time.time()

def is_taxicab(num):
    croot = round(num**(1/3) + 0.5)
    seq = range(1, croot)
    comb = [x for x in itertools.combinations(seq, 2) if x[0]**3 + x[1]**3 == num]
    
    if len(comb) >= 2:
        return "Y"
    else:
        return "N"

result = input.copy()
result['Answer Expected'] = result['Numbers'].apply(is_taxicab)

end_time = time.time()
execution_time = end_time - start_time
print("Execution time:", execution_time, "seconds")
# Execution time: 0.012981891632080078 seconds

print(result['Answer Expected'].equals(test['Answer Expected'])) # True